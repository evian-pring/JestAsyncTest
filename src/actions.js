const funcOne = async () => {
  console.log('One');
};

const funcTwo = async () => {
  console.log('Two');
};

export default { asyncOne: funcOne, asyncTwo: funcTwo };
