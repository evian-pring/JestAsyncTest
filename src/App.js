import React, { Component } from 'react';
import { View } from 'react-native';
import actions from './actions';

export class App extends Component {
  async componentDidMount() {
    console.log('In CDM');
    await actions.asyncOne();
    await actions.asyncTwo();
    console.log('Finished async');
  }

  render() {
    return <View />;
  }
}
