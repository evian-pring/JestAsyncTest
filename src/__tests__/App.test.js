import React from 'react';
import { App } from '../App';
import renderer from 'react-test-renderer';

import actions from '../actions';

const spyOne = jest.spyOn(actions, 'asyncOne');
const spyTwo = jest.spyOn(actions, 'asyncTwo');

describe('App ', () => {
  test('does async stuff in expected order', async () => {
    console.log('Starting test');
    const tree = await renderer.create(<App />);
    console.log('About to expect');
    expect(spyOne).toHaveBeenCalled();
    console.log('Expect one to have been called');
    expect(spyTwo).toHaveBeenCalled();
    console.log('Expect two to have been called');
    expect(tree).toMatchSnapshot();
  });
});
