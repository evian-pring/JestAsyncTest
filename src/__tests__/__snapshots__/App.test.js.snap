// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App  does async stuff in expected order 1`] = `<View />`;
